### PR TITLE
Implement Expo React Native

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR implements Expo 49.0.23 with React Native version 0.72.15

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Run yarn install
2. Go to apps/storybook-react-native
3. Run yarn ios

### **After**
Simulator Screenshot - iPhone 15 Pro Max - 2024-11-04 at 14 12 24

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs)
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by brianacnguyen** [comment]
https://github.com/SocketSecurity ignore-all

**Comment by brianacnguyen** [comment]
@SocketSecurity ignore-all

File: apps/storybook-react-native/src/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  text: string;
+};
+
+export const MyButton = ({ onPress, text }: MyButtonProps) => {
+  return (
+    <TouchableOpacity>
+      <Text>{text}</Text>
+    </TouchableOpacity>
+  );
+};
```
Comment (georgewrmarshall) [inline apps/storybook-react-native/src/Button.tsx:15]: non-blocking: doesn't look like this file is being used

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Left a couple non-blocking comments. I also noticed `Button.tsx` isn't used and it would be good to update the README to reflect running expo. Created an issue here https://github.com/MetaMask/metamask-design-system/issues/82

File: .yarnrc.yml [hunk 2]
```diff
@@ -15,3 +15,5 @@ nodeLinker: node-modules
 plugins:
   - path: .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs
     spec: 'https://raw.githubusercontent.com/LavaMoat/LavaMoat/main/packages/yarn-plugin-allow-scripts/bundles/@yarnpkg/plugin-allow-scripts.js'
+
+nmHoistingLimits: none
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply .yarnrc.yml:19]: This just shows that there's no hoisting limit

File: apps/storybook-react-native/src/Button.tsx (hunk 1 above)
Comment (brianacnguyen) [reply apps/storybook-react-native/src/Button.tsx:15]: will be for storybook. just setting it up right now

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**